### PR TITLE
Avoid unnecessary call to ByteBuf.readableBytes() from ByteToMessageDecoder

### PR DESCRIPTION
**Pull request by casidiablo**

### CODE REVIEW

**Comment by netkins** [comment]
**AUTOMATED MESSAGE for ADMINS**:<br>Please verify and accept the pull request.<br>To accept, say `@netkins accept`<br>To build again, say `@netkins build`<br>To whitelist the author, say `@netkins whitelist`

File: codec/src/main/java/io/netty/handler/codec/ByteToMessageDecoder.java [hunk 1]
```diff
@@ -196,8 +196,8 @@ protected ByteBuf internalBuffer() {
     @Override
     public final void handlerRemoved(ChannelHandlerContext ctx) throws Exception {
         ByteBuf buf = internalBuffer();
-        int readable = buf.readableBytes();
         if (buf.isReadable()) {
+            int readable = buf.readableBytes();
```
Comment (normanmaurer) [inline codec/src/main/java/io/netty/handler/codec/ByteToMessageDecoder.java]: Actually even better would be to do:

```java
int readable = buf.readableBytes();
if (readable > 0) {
    ....
}
```

This way you have only one method call in both cases. Would you please update the PR with this and ping me again ?

**Comment by casidiablo** [comment]
You are right. Just pushed a new commit.

**Comment by normanmaurer** [comment]
@casidiablo awesome.. just three things..

- Please squash the commits into one
- Update the commit message to match our template: http://netty.io/wiki/writing-a-commit-message.html
- Sign the ICLA: https://docs.google.com/forms/d/1GKuaUw10DfPzPy_XBnp3Wk_OLYnA4-Kg16wSiUkO3-A/viewform?formkey=dHBjc1YzdWhsZERUQnhlSklsbG1KT1E6MQ

After you completed these steps I will cherry-pick the change :)

**Comment by casidiablo** [comment]
Done

**Comment by normanmaurer** [comment]
@casidiablo thanks a lot! Cherry-picked into 4.0 (37cca81f1c47e28cee6610b549017f33e3b8e38c), 4.1 (b8d26d1da3f203033e1f5814d9b72582376cb0b5) and master (0b4d88bde5ec75afb2855fab60068309337a4c80)

**Pull request closed without merging**